test/test_memory.cpp

```diff
@@ -0,0 +1,79 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <regex>
+
+#include <fusion.h>
+#include <inlining.h>
+#include <ir/utils.h>
+#include <ops/alias.h>
+#include <ops/arith.h>
+#include <ops/utils.h>
+#include <options.h>
+#include <test/utils.h>
+#include <test/validator.h>
+#include <type.h>
+
+namespace nvfuser {
+
+class MemoryTest : public NVFuserTest {};
+
+TEST_F(MemoryTest, LoadCache) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* tv0 = makeContigTensor(1);
+  fusion.addInput(tv0);
+  TensorView* tv1 =
+      ops::newValLike(tv0, tv0->getDataType().value())->as<TensorView>();
+  IrBuilder::create<LoadStoreOp>(
+      LoadStoreOpType::Set, tv1, tv0, CacheOp::AllLevels);
+  TensorView* tv2 = add(tv1, IrBuilder::create<Val>(1.0));
+  TensorView* tv3 = set(tv2);
+  fusion.addOutput(tv3);
+
+  tv1->split(0, 4);
+  tv1->split(0, 32);
+  TransformPropagatorWithCheck propagator(tv1);
+  MaxRootDomainInfoSpanningTree(tv1).traverse(&propagator);
+
+  // Parallelize LoadStoreOps. Other TensorViews don't support vectorization.
+  tv1->axis(0)->parallelize(ParallelType::BIDx);
+  tv1->axis(1)->parallelize(ParallelType::TIDx);
+  tv1->axis(2)->parallelize(ParallelType::Vectorize);
+  scheduler_utils::parallelizeAllLike(tv1, {tv3});
+
+  inlineMost();
+
+  at::Tensor input = at::randn(
+      {1024}, at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0));
+  at::Tensor expected_output = input + 1.0f;
+
+  FusionExecutor fe;
+  fe.setSaveCompiledBinaryFlag(true);
+  {
+    DebugDumpOptionsGuard debug_dump_options_guard;
+    DebugDumpOptionsGuard::getCurOptions().set(DebugDumpOption::Ptx);
```

Comment:
With this line, any run of `bin/nvfuser_tests` produces a `__tmp_kernel1.ptx` file in cwd, which can be annoying. I could change PTX dumping to dump to text instead (similar to FusionIR dumping etc), but not sure if that's going too far.

Reply:
cc @naoyam

Reply:
Yeah, that's a little annoying. Can we just remove the file?

Reply:
What do you mean? I don't think there's programmatic way to get the filename. It's not always `__tmp_kernel1.ptx` -- sometimes I saw `__tmp_kernel3496.ptx` or something.

Reply:
The number should be the kernel ID, so we should be able to know the file name or add an interface to query it if needed.

Reply:
Done.